src/services/api.service.js

```diff
@@ -12,19 +13,162 @@ export function get (endpoint) {
     });
 }
 
-export function post (endpoint, payload = {}) {
+export function getFavoriteBooks () {
     return new Promise((resolve) => {
-        fetch(`../data/${endpoint}.json`, {
-            method: 'POST',
-            headers: {
-                'Content-Type': 'application/json'
-            },
-            body: JSON.stringify(payload),
-        })
-            .then(() => {
+        fetch(`../data/books.json`)
+            .then((response) => response.json())
+            .then((json) => {
+                setTimeout(() => {
+                    resolve(json.data.filter((book) => book.favorite));
+                }, serverLatency);
+            });
+    });
+}
+
+export function getBookByIsbn (isbn) {
+    return new Promise((resolve, reject) => {
+        fetch(`../data/books.json`)
+            .then((response) => response.json())
+            .then((json) => {
                 setTimeout(() => {
-                    resolve();
+                    const book = json.data.find((book) => book.isbn === isbn);
+                    if (!book) {
+                        reject(new Error('Book not found'));
+                    }
+                    resolve(book);
                 }, serverLatency);
             });
     });
 }
+
+export function createBook (book) {
+    return new Promise((resolve, reject) => {
+        getBookByIsbn(book.isbn)
+            .catch(() => {
+                validateBook(book).catch((error) => reject(error));
+
+                const defaultValues = {
+                    pageCount: -1,
+                    publishedDate: '',
+                    thumbnailUrl: 'https://picsum.photos/150/190',
+                    shortDescription: '',
+                    longDescription: '',
+                    categories: [],
+                    favorite: false,
+                };
+                const bookToCreate = {
+                    ...defaultValues,
+                    ...book,
+                };
+
+                fetch(`../data/books.json`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify(bookToCreate),
+                }).then(() => {
+                    setTimeout(() => {
+                        resolve();
```

Comment:
do we want to `resolve(bookToCreate)` so like most endpoints returns the book from the server?